add check to disallow setting a column as NOT NULL (#101)
sbdchd/squawk: docs/docs/adding-not-nullable-field.md

```

Add the column as nullable, add a check constraint as `NOT VALID` that verifies the column is not null, backfill the column so it no longer contains null values, validate the constraint to verify existing rows are valid.

See ["How not valid constraints work"](constraint-missing-not-valid.md#how-not-valid-validate-works) for more information on adding constraints as `NOT VALID`.

### setting an existing column as non-nullable

Instead of:

```sql
ALTER TABLE "core_recipe" ALTER COLUMN "foo" SET NOT NULL;
```

Use:

```sql
ALTER TABLE "core_recipe" ADD CONSTRAINT foo_not_null
    CHECK ("foo" IS NOT NULL) NOT VALID;
-- backfill column so it's not null
ALTER TABLE "core_recipe" VALIDATE CONSTRAINT foo_not_null;
```

> Add a check constraint as `NOT VALID` that verifies the column is not null, backfill the column so it no longer contains null values, validate the constraint to verify existing rows are valid.

Reviewer: ```suggestion
Add a check constraint as `NOT VALID` to verify new rows and updates are not null, backfill the column so it no longer contains null values, validate the constraint to verify existing rows are valid.
```
